test/server/http/admin_filter_test.cc

```diff
@@ -0,0 +1,88 @@
+#include "server/http/admin_filter.h"
+
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::InSequence;
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Server {
+
+class AdminFilterTest : public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  AdminFilterTest()
+      : admin_server_callback_func_(createAdminServerCallback()),
+        filter_(admin_server_callback_func_), request_headers_{{":path", "/"}} {
+    filter_.setDecoderFilterCallbacks(callbacks_);
+  }
+
+  typedef std::function<Http::Code(absl::string_view path_and_query,
+                                   Http::ResponseHeaderMap& response_headers,
+                                   Buffer::OwnedImpl& response, AdminFilter& filter)>
+      AdminServerCallbackFunction;
+
+  NiceMock<MockInstance> server_;
+  Stats::IsolatedStoreImpl listener_scope_;
+  AdminServerCallbackFunction admin_server_callback_func_;
+  AdminFilter filter_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  Http::TestRequestHeaderMapImpl request_headers_;
+
+  AdminServerCallbackFunction createAdminServerCallback() {
```

Comment:
You don't have to generate this function here; you can just make a static method:
```
static Http::Code adminServerCallback(absl::string_view path_and_query, Http::ResponseHeaderMap& response_headers,
              Buffer::OwnedImpl& response, AdminFilter& filter) {
  ...
  return Http::Code::OK;
}
```